### PR TITLE
Add more examples and cleanup docs for auth

### PR DESCRIPTION
**Pull request by sethvargo**
/cc @jefferai

### CODE REVIEW

File: command/auth.go [hunk 1]
```diff
@@ -297,29 +297,33 @@ func (c *AuthCommand) Synopsis() string {
...
-  "vault auth -method=github -path=github-private token=<github_token>"
-  The value of the "path" flag will be supplied to auth providers
-  as the "mount" option in the payload to specify the mount point.
-  See the "-method-help" for more info.
+  By default, the -method is assumed to be token. If not supplied via the
+  command-line, a prompt for input will be shown. If the authentication
+  information is "-", it will be read from stdin.
+
+  The -method option allows alternative authentication methods to be used,
+  such as userpass, oauth, or TLS certificates. For these, additional values
```
Comment (vishalnayak) [inline command/auth.go]: I would remove `OAuth` and use `GitHub token` instead.

**Comment by vishalnayak** [comment]
One minor comment, otherwise LGTM!

**Comment by jefferai** [comment]
LGTM with Vishal's suggested change.

**Comment by jefferai** [comment]
Made the change, merging.

**Pull request merged**